prusti-common/src/vir/ast/expr.rs

```diff
@@ -1198,7 +1198,7 @@ impl Expr {
         }
         let typaram_substs = match (&target, &replacement) {
             (Expr::Local(tv, _), Expr::Local(rv, _)) => {
-                if tv.typ.is_ref() && rv.typ.is_ref() {
+                if tv.typ.is_ref() && rv.typ.is_ref() && !replacement.local_type().starts_with("Array$") {
```

Comment:
Generics are messy… What exactly goes wrong if you don't have this extra condition? Does it complain somewhere later that a type is not a place?

Reply:
not sure anymore. removed it, array testcases still pass. i'll do a full run of the testsuite and see, maybe it's not needed (anymore?) after all.

Reply:
tests look good :-)